mizani/breaks.py

```diff
@@ -141,6 +142,10 @@ def __call__(self, limits):
         _min = int(np.floor(rng[0]))
         _max = int(np.ceil(rng[1]))
 
+        # prevent overflow
+        if float(base)**_max > sys.maxsize:
+           base = float(base)
```

Comment:
This could be made nicer with the newly introduced assignment expressions: https://www.python.org/dev/peps/pep-0572/ but that requires Python 3.8.
```
if (fbase := float(base))**_max > sys.maxsize:
    base = fbase
```